index.js

```diff
@@ -163,6 +71,156 @@ class ShareSheet extends React.Component<Props> {
+        })
+        .then(hasPermission => {
+          if (hasPermission) {
+            return Promise.resolve(PermissionsAndroid.RESULTS.GRANTED);
+          }
+          return PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE);
+        })
+        .then(result => {
+          if (result === PermissionsAndroid.RESULTS.GRANTED) {
+            return resolve();
+          }
+          return Promise.reject(new Error('Write Permission not available'));
+        })
+        .catch(e => reject(e));
+    });
+  }
+  return Promise.resolve();
+};
+
+class RNShare {
+  static open(options: Options): Promise<OpenReturn> {
+    return new Promise((resolve, reject) => {
+      requireAndAskPermissions(options)
+        .then(() => {
+          if (Platform.OS === 'ios') {
+            if (options.urls) {
+              // Handle for multiple file share
+              NativeModules.RNShare.open(
+                options,
+                error => {
+                  return reject({ error: error });
+                },
+                (success, activityType) => {
+                  if (success) {
+                    return resolve({
+                      app: activityType,
+                    });
+                  } else if (options.failOnCancel === false) {
+                    return resolve({
+                      dismissedAction: true,
+                    });
+                  } else {
+                    reject({ error: 'User did not share' });
+                  }
+                },
+              );
+            } else {
+              // Handle for single file share
+              ActionSheetIOS.showShareActionSheetWithOptions(
+                options,
+                error => {
+                  return reject({ error: error });
+                },
+                (success, activityType) => {
+                  if (success) {
+                    return resolve({
+                      app: activityType,
+                    });
+                  } else if (options.failOnCancel === false) {
+                    return resolve({
+                      dismissedAction: true,
+                    });
+                  } else {
+                    reject({ error: 'User did not share' });
+                  }
+                },
+              );
+            }
+          } else {
+            NativeModules.RNShare.open(
+              options,
+              e => {
+                return reject({ error: e });
+              },
+              e => {
+                resolve({
+                  message: e,
+                });
+              },
+            );
+          }
+        })
+        .catch(e => reject(e));
+    });
+  }
+
+  static shareSingle(options: Options): Promise<ShareSingleReturn> {
+    if (Platform.OS === 'ios' || Platform.OS === 'android') {
+      return new Promise((resolve, reject) => {
+        requireAndAskPermissions(options)
+          .then(() => {
```

Comment:
can't you use `await` and `catch`? It is so much cleaner then `then` and `catch`

Reply:
for using await and async, projects need to use newer babel. Some older projects may not have them implemented, so I just tried to avoid that.

Reply:
this is not true

react-native supports async await out of the box in all projects

Reply:
I thought it was not available in 0.30.x versions of react-native. If it was there, thats good.